### ISSUE TITLE
CloudNative.CloudEvents.RabbitMq

### ISSUE THREAD

**Issue by antiblue** [cloudevents/sdk-csharp#310]
Hi, I worked on an AMQP 0.9.1 protocol binding using the [RabbitMq.Client](https://www.rabbitmq.com/client-libraries/dotnet-api-guide#overview).

I followed the ["RabbitMQ Protocol Binding for CloudEvents" specification](https://github.com/knative-extensions/eventing-rabbitmq/blob/main/cloudevents-protocol-spec/spec.md)
and looked (of course) at how the [AMQP 1.0 bindings](https://github.com/cloudevents/sdk-csharp/tree/main/src/CloudNative.CloudEvents.Amqp) here do things. 

I would like to create a PR to get this integrated into the sdk-csharp.

**Comment by jskeet** [comment]
That sounds fine, yes. Maybe leave this issue open for any discussion about aspects that there isn't code for yet, and create a PR to add the library (and tests).

**Comment by jskeet** [comment]
(If the new library can be independent of the existing Amqp library, that would be handy in terms of diamond dependencies etc, but let me know if that doesn't work out. I really don't know a lot about this area I'm afraid.)

**Comment by antiblue** [comment]
Yes, I could not use the existing library and the new one does not depend on `CloudNative.CloudEvents.Amqp` or `AMQPNetLite`

**Comment by antiblue** [comment]
Hi @jskeet I face a strange issue.
I am unable to restore packages on [my branch](https://github.com/antiblue/sdk-csharp/tree/feature/T-310-rabbitmq) because he basically tries to find itself.

The error message is
`[NU1101] Unable to find package CloudNative.CloudEvents.RabbitMq. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages, nuget.org at (0:0)`

Any ideas?

**Comment by jskeet** [comment]
I'll see if I can find time to pull your branch and reproduce, but I make no guarantees I'm afraid.

**Comment by jskeet** [comment]
Got it - the problem is that it's trying to compare the current code with a previous version of the same package (for backward compatibility). Add this into the property group:

```xml
<EnablePackageValidation>false</EnablePackageValidation>
```

... then do a full rebuild. It should work at that point.
